### DISCUSSION TITLE
Packaging CIEL for NixOS

### DISCUSSION THREAD

**Discussion by tani** [ciel-lang/CIEL#82] [CIEL Ideas]
Hi there!

Nix is ​​a package manager on par with apt-get, yum (dnf), and brew.

Like the GUIX installation supported by CIEL, NixOS is a general-purpose package manager that can be used with the dedicated OS or common Linux distributions. Nix has recently been adopted by Lem, and its use may increase in the CL community.

@Comamoca and I are working to package CIEL into Nix.

Now, there are various difficulties in packaging CIEL in Nix.

- Nix can only install LISP systems uploaded to Quicklisp
  - `file-finder` and `moira/light` cannot be installed
- ASDF and Nix seem to have a strong connection on each other
  - When you install a LISP system with Nix, the path is automatically passed to ASDF
  - But ASDF is still the old 3.3.1

I think both of these issues will be resolved with time (SBCL will update the ASDF version, and Quicklisp will update),
but for now it seems necessary to hack Nix and package CIEL.

`file-finder` and `moira/light` are already being solved (including fixes to Nix itself).

However, the second issue of the old ASDF version has not yet been solved.

This is because it is not easy to update ASDF that is bound to Nix.

Although this problem is specific to Nix, the fact that ASDF that comes with SBCL, which is the de facto standard, doesn't build leaves the possibility of it potentially causing problems for other installations in the future. Is it possible to use features or something to make the limited-feature version of CIEL work with older ASDFs? Please consider adding something like `#-asdf.3.3.1` to `local-nicknames`.

Thanks for a great project!

**Comment by vindarel** [comment]
Thank you for your work. Packaging is very important for CIEL and I am thrilled that many of you are contributing.

`file-finder` and `moira/light` are now in the latest Quicklisp release (Oct, 2024).

> Is it possible to use features or something to make the limited-feature version of CIEL work with older ASDFs? Please consider adding something like #-asdf.3.3.1 to local-nicknames.

\+1, I'll get to it and I accept PRs in the meantime.
